src/inc/quic_platform_linux.h

```diff
@@ -633,6 +633,29 @@ typedef struct QUIC_THREAD_CONFIG {
     void* Context;
 } QUIC_THREAD_CONFIG;
 
+#ifdef QUIC_USE_CUSTOM_THREAD_CONTEXT
+
+//
+// Extension point that allows additional platform specific logic to be executed
+// for every thread created. The platform must define QUIC_USE_CUSTOM_THREAD_CONTEXT
+// and implement the QuicThreadCustomStart function. QuicThreadCustomStart MUST
+// call the Callback passed in. QuicThreadCustomStart MUST also free
+// CustomContext before returning.
+//
+
+typedef struct QUIC_THREAD_CUSTOM_CONTEXT {
+    LPTHREAD_START_ROUTINE Callback;
+    void* Context;
+} QUIC_THREAD_CUSTOM_CONTEXT;
+
+void*
```

Comment:
I feel like we should have a `typedef QUIC_CUSTOM_THREAD_RETURN_TYPE void*` here and `typedef QUIC_CUSTOM_THREAD_RETURN_TYPE DWORD` in windows. That way the implementing code can just say `QUIC_CUSTOM_THREAD_RETURN_TYPE QuicThreadCustomStart` for their implementation, rather then needing to ifdef. Since in most cases, the other side of this will not be operating system specific.